File: internal/models/confidence.go
package models

// PropagationStep records one step of confidence propagation.
type PropagationStep struct {
	NodeID             string   `json:"node_id"`
	ComputedConfidence float64  `json:"computed_confidence"`
	SourceNodes        []string `json:"source_nodes"`
	Formula            string   `json:"formula"`
}

// ConfidenceResult is the output of confidence propagation.
type ConfidenceResult struct {
	OverallConfidence float64            `json:"overall_confidence"`
	NodeConfidences   map[string]float64 `json:"node_confidences"`
	PropagationPath   []PropagationStep  `json:"propagation_path"`
}

File: src/explainable_engine/models/confidence.py
from pydantic import BaseModel


class PropagationStep(BaseModel):
    node_id: str
    computed_confidence: float
    source_nodes: list[str]
    formula: str


class ConfidenceResult(BaseModel):
    overall_confidence: float
    node_confidences: dict[str, float]
    propagation_path: list[PropagationStep]
